packages/next/server/next-server.ts

```diff
@@ -239,7 +239,7 @@ export default class Server {
 
     this.distDir = join(this.dir, this.nextConfig.distDir)
     this.publicDir = join(this.dir, CLIENT_PUBLIC_FILES_PATH)
-    this.hasStaticDir = !minimalMode && fs.existsSync(join(this.dir, 'static'))
+    this.hasStaticDir = !minimalMode && this.getHasStaticDir()
```

Comment:
Maybe instead of a method we can make this a property that must be initialized by the constructor of the implementing class.

Reply:
Yes let's do that in follow-up PRs! Gonna ship this one first so it will unblock us.